### PR TITLE
Add community K-hop implementation

### PR DESCRIPTION
**Pull request by AidenPearce-ZYQ**

### CODE REVIEW

File: procedures/community/README.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
 
- In this directory, The procedures are contribution from the commmunity.
\ No newline at end of file
+ In this directory, The procedures are contribution from the commmunity.
+
+## K-hop Algorithm
+The K-hop algorithm is used to find all nodes within K hops from a given starting node in a graph. It is particularly useful in social network analysis, recommendation systems, and network topology analysis.
+
+### Contributor
+This algorithm was contributed by AidenPearce-ZYQ, iwanttoknowwhy and ZhengHeber.
```
Comment (qishipengqsp) [inline procedures/community/README.md:9]: Add links to your Github homepages.

File: procedures/community/khop_core.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2024 Yingqi Zhao
+ * 
+ * @Function: Compute the k-hop algorithm.
```
Comment (qishipengqsp) [inline procedures/community/khop_core.cpp:4]: check and revise the file header

File: procedures/community/khop_standalone.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Yingqi Zhao
```
Comment (qishipengqsp) [inline procedures/community/khop_standalone.cpp:1]: DITTO about the file header

File: procedures/CMakeLists.txt [hunk 4]
```diff
@@ -81,6 +81,7 @@ add_standalone(wlpa)
 add_standalone(subgraph_isomorphism)
 add_standalone(sybilrank)
 add_standalone(leiden)
+add_stanalone(khop)
```
Comment (qishipengqsp) [inline procedures/CMakeLists.txt:84]: the add_standalone command here is to compile the procedures under algo_cpp directory instead of community

Comment (AidenPearce-ZYQ) [reply procedures/CMakeLists.txt:84]: We have fixed it. Thanks!

**Pull request merged**